javers-core/src/main/java/org/javers/core/metamodel/type/TypeMapperState.java

```diff
@@ -155,15 +161,20 @@ private void addFullMapping(Type javaType, JaversType newType){
     /**
      * must be called within synchronized block
      */
-    private JaversType infer(Type javaType) {
+    private JaversType infer(Type javaType, boolean asShallowReference) {
```

Comment:
wydaje się, że można to ograć w samym GraphBuilderze i/lub ChangeAppenderach bez dodawania flag do TypeMappera. Dobrze by było zachować model, że 1 java type jest mapowany na 1 javers type. Pomyślę jak to zrobić